test/test_dynamic_shapes.py

```diff
@@ -554,6 +555,70 @@ def test_expect_true_double_digits(self):
         self.assertTrue(expect_true(sum(ia) == 20))
         self.assertEqual(len(shape_env.deferred_runtime_asserts[ia[-1].node.expr]), 1)
 
+    def test_expect_true_refine_range(self):
+        shape_env = ShapeEnv()
+        for rel in [lambda x: x > 4, lambda x: 4 < x, lambda x: x >= 5, lambda x: 5 <= x]:
```

Comment:
Nit: probably want to add some 'with self.subTest' in this for loop to make debugging easier.

Reply:
bah, then I need to name the lambdas, someone else can add it if they need it LOL